tests/cancelled.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2018 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blackbox
+
+import (
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+)
+
+var (
+	testCancelled     bool
+	testCancelledLock sync.Mutex
+)
+
+func init() {
+	signalChannel := make(chan os.Signal, 1)
+	signal.Notify(signalChannel, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		sig := <-signalChannel
+		switch sig {
+		case os.Interrupt, syscall.SIGTERM:
+			cancelTheTests()
+		}
+	}()
+}
+
+func testsAreCancelled() bool {
+	testCancelledLock.Lock()
+	cancelled := testCancelled
+	testCancelledLock.Unlock()
+	return cancelled
+}
+
+func cancelTheTests() {
```

Comment:
Print something that indicates what's going on? Otherwise it will just look like nothing happened.

Reply:
I'll print something, but imo if a user hits Ctrl+C they'll see the `^C` and will probably know that they hit it. Plenty of other things (e.g. `./build`, `curl`, `git`, ...) don't explain what Ctrl+c does when they're killed.